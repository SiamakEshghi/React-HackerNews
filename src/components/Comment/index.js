import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Spinner from 'components/Spinner';
import 'components/style.css';

function Comment(props) {
    const { commentList, loading } = props;

    if(loading) {
        return (
            <div className="spinner">
                <Spinner />
            </div>
        );
    }
   const renderComment = () => {
        return (
            commentList.map((comment) => <h5 key={comment.id}>{`* ${comment.text}`}</h5>)
        );
    }
    return (
        <div className="container">
            <h3>Comments</h3>
            <div className="list">
                {renderComment()}
            </div>
        </div>
    );
};

const mapStateToProps = ({ reducer }) => {
    const { commentList, loading } = reducer;
    return { commentList, loading };
}

Comment.prototype = { commentList: PropTypes.array.isRequired }
export default connect(mapStateToProps)(Comment);