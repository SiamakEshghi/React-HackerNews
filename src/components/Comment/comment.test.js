import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import reduxThunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import Comment from 'components/Comment';

let wrapper, store;
const middlewares = [reduxThunk];
const mockstore = configureStore(middlewares);
const initialState = {reducer: {topStoryList: [], commentList: [{id: 1, text: 'text1'}, {id: 2, text: 'text2'}] }}

beforeEach(() => {
    store = mockstore(initialState);
    wrapper = mount(<Provider store={store}><Comment/></Provider>);
});
afterEach(() => {
    wrapper.unmount();
});
it('Comment snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
})