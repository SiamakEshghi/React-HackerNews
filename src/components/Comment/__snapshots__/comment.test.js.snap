// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment snapshot test 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Comment)>
    <Comment
      commentList={
        Array [
          Object {
            "id": 1,
            "text": "text1",
          },
          Object {
            "id": 2,
            "text": "text2",
          },
        ]
      }
      dispatch={[Function]}
    >
      <div
        className="container"
      >
        <h3>
          Comments
        </h3>
        <div
          className="list"
        >
          <h5
            key="1"
          >
            * text1
          </h5>
          <h5
            key="2"
          >
            * text2
          </h5>
        </div>
      </div>
    </Comment>
  </Connect(Comment)>
</Provider>
`;
